### PR TITLE
feat: individual model clipping planes

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/REV-682

## Description :pencil:

Adds the ability to set clipping planes individually for each model, in addition to globally. The global and "local" clipping planes will both be evaluated, resulting in the intersection of the two volumes.

In addition:
- Deprecated `set/getClippingPlanes` in the viewer, added `set/getGlobalClippingPlanes` instead
- Disable picking for clipped points in point clouds
- Expanded clipping plane UI

## How has this been tested? :mag:

Tested our local models (both CAD and pointcloud) and see that clipping planes smoothly interact when set from the UI.

Point cloud picking: 
- Open https://localhost:3000/?project=3d-test&env=greenfield-3d&modelId=7394378315995345&revisionId=36918959705705
- Wait for object load, click fit to camera when done
- Move the camera so that one of the big towers are hidden behind another
- Clip the closest with clipping planes, ensuring the one behind is now visible
- Click it and see that it's highlighted

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [ ] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [x] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/packages/api/src/public/migration/Cognite3DViewer.ts [hunk 1]
```diff
@@ -928,38 +928,55 @@ export class Cognite3DViewer {
...
+   * viewer.setGlobalClippingPlanes([plane]);
    * ```
    * ```js
    * // Disable clipping planes
-   *  viewer.setClippingPlanes([]);
+   *  viewer.setGlobalClippingPlanes([]);
    * ```
    */
-  setClippingPlanes(clippingPlanes: THREE.Plane[]): void {
+  setGlobalClippingPlanes(clippingPlanes: THREE.Plane[]): void {
```
Comment (christjt) [inline viewer/packages/api/src/public/migration/Cognite3DViewer.ts:954]: Concider spear oparator here to allow for 1 instance or list.

File: viewer/packages/cad-model/src/wrappers/CogniteCadModel.ts [hunk 2]
```diff
@@ -335,6 +335,14 @@ export class CogniteCadModel implements CdfModelNodeCollectionDataProvider {
     return this.cadNode.getModelTransformation(out);
   }
 
+  /**
+   * Sets the clipping planes for this model. They will be combined with the
+   * global clipping planes.
+   */
+  setModelClippingPlanes(clippingPlanes: THREE.Plane[]): void {
```
Comment (christjt) [inline viewer/packages/cad-model/src/wrappers/CogniteCadModel.ts:342]: Concider spread operator here

File: viewer/packages/pointclouds/src/CognitePointCloudModel.ts [hunk 3]
```diff
@@ -227,6 +227,14 @@ export class CognitePointCloudModel {
     this.pointCloudNode.pointShape = shape;
   }
 
+  /**
+   * Sets the clipping planes for this model. They will be combined with the
+   * global clipping planes.
+   */
+  setModelClippingPlanes(clippingPlanes: THREE.Plane[]): void {
```
Comment (christjt) [inline viewer/packages/pointclouds/src/CognitePointCloudModel.ts:234]: Concider spread operator here.

**Review by christjt — APPROVED** [review]
LGTM, only small suggestion. Also, can you create a task for storing / restoring local clipping planes in view state?

File: viewer/packages/api/src/public/migration/Cognite3DViewer.ts (hunk 1 above)
Comment (haakonflatval-cognite) [reply viewer/packages/api/src/public/migration/Cognite3DViewer.ts:954]: After some discussion, we concluded that the spread/rest operator isn't an ideal fit here. Will leave as is.

**Pull request merged**